deps: Remove dependency on github.com/openshift/library-go

akrejcir opened the pull request:
### What this PR does
We only used 3 utility functions from this library, they can be easily copied directly to the codebase to remove a dependency.

This is a copy of https://github.com/kubevirt/kubevirt/pull/13997 because it could not be reopened.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
None
```

akrejcir committed:
deps: Remove dependency on github.com/openshift/library-go

Kubevirt only used 3 utility functions from this library,
they can be easily copied directly to the codebase to
remove a dependency.

It a different PR, when this dependency is updated to
the latest version, it would fail FOSSA check.

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir committed:
tests: add unit tests for new packages

Added unit tests for:
- pkg/util/naming
- pkg/virt-operator/resources/apply/resourcemerge

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir committed:
lint: enable linter for new packages

Enabled lint for:
- pkg/virt-operator/resource/apply/resourcemerge
- pkg/util/naming

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir committed:
cleanup: Refactor virt-operator/resources/apply/generations.go

Small refactoring of the file.

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir commented:
/retest

akalenyu reviewed (commented):
~~I would reopen https://github.com/kubevirt/kubevirt/pull/13997 instead, the FOSSA thing is probably a false positive.~~

I don't think it's an issue normally to consume these libraries

akrejcir commented:
I've force pushed new commits to the same branch that was used in the previous PR https://github.com/kubevirt/kubevirt/pull/13997, so now I don't see a way to reopen it.

The FOSSA failure was solved in another PR: https://github.com/kubevirt/kubevirt/pull/15471

akalenyu commented:
> I've force pushed new commits to the same branch that was used in the previous PR #13997, so now I don't see a way to reopen it.
> 
> The FOSSA failure was solved in another PR: #15471

Sorry I meant reopen https://github.com/kubevirt/kubevirt/pull/15471, which is merged, so I'm confused about why this is needed at all. I really think it's legitimate to have this dep? @0xFelix wdyt

akrejcir committed:
cleanup: resourcemerge: Move output parameter to return value

It is unusual to have an output parameter in golang.

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir committed:
cleanup: resourcemerge: Remove unused logic from helper functions

The mergeMap and mergeOwnerRefs functions supported removing elements,
but it was never used.
I've tested it by adding "panic" to the relevant code paths,
and running `make cluster-sync`. It did not panic.

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir committed:
cleanup: resourcemerge: Simplify code

Minor simplification of code.

Signed-off-by: Andrej Krejcir <akrejcir@redhat.com>

akrejcir commented:
/retest

akrejcir commented:
Closing, because there was no progress on this for a long time.

/close
